### PR TITLE
feat(datepicker): max date range for daterangepicker (#5562)

### LINKED ISSUE #5562: Daterangepicker: Max date range feature
**Is your feature request related to a problem? Please describe.**
We would like to be able to enforce that a selected date range is not larger then for example "6 months". 

**Describe the solution you'd like**
Setting this value would allow you to select any date between the existing `minDate`/`maxDate`, after selecting the first date the dates that don't fall within the `maximumDateRange` would be disabled

**Describe alternatives you've considered**
I tried to modify the `minDate`/`maxDate` after selecting the first date of the dateRange. But I didn't see any callback or method that responds to selecting the first date

### PR DESCRIPTION
**Pull request by arvindm95**
closes #5562 

# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated tests.
 - [x] added/updated API documentation.
 - [x] added/updated demos.

### CODE REVIEW

File: src/datepicker/themes/bs/bs-daterangepicker-container.component.ts [hunk 1]
```diff
@@ -155,4 +158,10 @@ export class BsDaterangepickerContainerComponent extends BsDatepickerAbstractCom
     this._store.dispatch(this._actions.selectRange(this._rangeStack));
   }
 
+  setMaxDateRangeOnCalendar(currentSelection: Date) {
```
Comment (IraErshova) [inline src/datepicker/themes/bs/bs-daterangepicker-container.component.ts]: Could you please add typing for returning data?

Comment (arvindm95) [reply src/datepicker/themes/bs/bs-daterangepicker-container.component.ts]: Yes, Done

**Comment by daniloff200** [comment]
Well, I see, that right now bunch of jobs on travis failed, I expect, it can be related to new locale, merged to the development, so, gonna update this branch with latest development

**Review by daniloff200 — DISMISSED** [review]
Looks good for me, let's wait for CI build

**Comment by daniloff200** [comment]
hey, @arvindm95 
you've got some conflicts with development branch (probably, from my latest merge of #5645) 
could you please fix them?

**Review by daniloff200 — DISMISSED** [review]

**Comment by arvindm95** [comment]
@dmitry-zhemchugov Can you please look into this? Having a dependency on this feature.

**Comment by arvindm95** [comment]
Can we please get this done soon? @daniloff200 @IraErshova @dmitry-zhemchugov Thanks!

**Comment by arvindm95** [comment]
@valorkin @daniloff200 @IraErshova @dmitry-zhemchugov please update this?

**Comment by arvindm95** [comment]
@valorkin  @daniloff200 @IraErshova @dmitry-zhemchugov any update on this?

**Comment by arvindm95** [comment]
@valorkin @dmitry-zhemchugov @daniloff200 Can we get this done?

**Comment by daniloff200** [comment]
@arvindm95 hey!

Sorry for the late response, the development was a little bit frozen, during the last month.
Right now I see, that you have some merge conflicts, and, afaik, there were some issues with the e2e tests in this PR
Can you please fix the conflicts, and later , I will check that case with e2e and the whole feature too

**Comment by arvindm95** [comment]
> @arvindm95 hey!
> 
> Sorry for the late response, the development was a little bit frozen, during the last month.
> Right now I see, that you have some merge conflicts, and, afaik, there were some issues with the e2e tests in this PR
> Can you please fix the conflicts, and later , I will check that case with e2e and the whole feature too

Hey @daniloff200 ,
Resolved the conflicts. Can you retry with the ci build?

**Comment by arvindm95** [comment]
Hi @daniloff200 , Sorry for pushing on this, there is a dependency that has to be closed soon. Can I expect this wrapped this week?

**Review by daniloff200 — APPROVED** [review]

**Pull request merged**